python/lsst/ts/watcher/base_hexapod_overcurrent_rule.py

```diff
@@ -0,0 +1,190 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["BaseHexapodOvercurrentRule"]
+
+import logging
+import types
+import typing
+
+import yaml
+from lsst.ts import salobj
+from lsst.ts.xml.enums.MTHexapod import ControllerState, EnabledSubstate, SalIndex
+from lsst.ts.xml.enums.Watcher import AlarmSeverity
+
+from .base_rule import AlarmSeverityReasonType, BaseRule, NoneNoReason
+from .remote_info import RemoteInfo
+
+
+class BaseHexapodOvercurrentRule(BaseRule):
+    """Base rule to monitor the main telescope hexapod overcurrent event.
+
+    Parameters
+    ----------
+    salIndex : enum `SalIndex`
+        Hexapod SAL index.
+    config : `types.SimpleNamespace`
+        Rule configuration, as validated by the schema.
+    log : `logging.Logger` or None, optional
+        Parent logger. (the default is None)
+    """
+
+    def __init__(
+        self,
+        salIndex: SalIndex,
+        config: types.SimpleNamespace,
+        log: logging.Logger | None = None,
+    ):
+        is_camera_hexapod = salIndex == SalIndex.CAMERA_HEXAPOD
+        prefix_warning = (
+            "MTCameraHexapodOvercurrent"
+            if is_camera_hexapod
+            else "MTM2HexapodOvercurrent"
+        )
+        remote_name = "MTHexapod"
+        remote_info = RemoteInfo(
+            remote_name,
+            salIndex.value,
+            callback_names=["evt_controllerState", "tel_electrical"],
+        )
+        super().__init__(
+            config,
+            f"{prefix_warning}.{remote_name}",
+            [remote_info],
+            log=log,
+        )
+
+        self._hexapod = "CameraHexapod" if is_camera_hexapod else "M2Hexapod"
+
+        self._controller_state = ControllerState.STANDBY
+        self._enabled_state = EnabledSubstate.STATIONARY
+
+        self._count = 0
+
+    @classmethod
+    def get_schema(cls) -> dict[str, typing.Any]:
+        schema_yaml = """
+            $schema: 'http://json-schema.org/draft-07/schema#'
+            description: Configuration for BaseHexapodOvercurrentRule rule.
+            type: object
+            properties:
+                threshold_current:
+                    description: >-
+                        Threshold of the current in ampere.
+                    type: number
+                    default: 4.0
+                max_count:
+                    description: >-
+                        Maximum count to check the hexapod current in idle.
```

Comment:
wouldn't it be easier to make this the "time window" (e.g. 10 minutes) instead?

Reply:
Updated in the commit: [c789acd](https://github.com/lsst-ts/ts_watcher/pull/94/commits/c789acde31876242324909796fb547c8ff2cf2f0). Thanks!